src/main/kotlin/ai/digital/integration/server/common/util/ProcessUtil.kt

```diff
@@ -30,36 +31,40 @@ class ProcessUtil {
             }
         }
 
-        fun execute(project: Project, exec: String, arguments: List<String>, logOutput: Boolean = true): String {
+        fun execute(
+            project: Project,
+            exec: String,
+            arguments: List<String>,
+            logOutput: Boolean = true
+        ): String {
             project.logger.lifecycle("About to execute `$exec ${arguments.joinToString(" ")}`")
+
+            val stdout = ByteArrayOutputStream()
+            val stderr = ByteArrayOutputStream()
+
+            val execOps = project.serviceOf<ExecOperations>()
+
+            execOps.exec {
+                executable = exec
+                args = arguments
+                standardOutput = stdout
+                errorOutput = if (logOutput) stderr else stdout
+                isIgnoreExitValue = false
+            }.assertNormalExitValue()
+
+            val output = stdout.toString(StandardCharsets.UTF_8)
+            val error = stderr.toString(StandardCharsets.UTF_8)
+
             if (logOutput) {
-                val stdout = ByteArrayOutputStream()
-                try {
-                    project.exec {
-                        args = arguments
-                        executable = exec
-                        standardOutput = stdout
-                    }
-                    val output = stdout.toString(StandardCharsets.UTF_8)
+                if (output.isNotBlank()) {
                     project.logger.lifecycle(output)
-                    return output
-                } finally {
-                    stdout.close()
                 }
```

Comment:
Why we have removed this else

Reply:
Added a new else with "" similar to earlier code 

else {
val stdout = NullOutputStream()
try {
project.exec {
args = arguments
executable = exec
standardOutput = stdout
errorOutput = stdout
}
} finally {
stdout.close()
}
}
return ""
}